src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -296,24 +315,37 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
     return results;
   }
 
-  static <K, V> void produceKeyValuesSynchronously(final String inputTopic,
-                                                   final List<KeyValue<K, V>> entities,
+  /**
+   * Like {@link IntegrationTestUtils#produceKeyValuesSynchronously(String, Collection, Properties)}, except for use
+   * with TopologyTestDriver tests, rather than "native" Kafka broker tests.
+   *
+   * @param topic              Kafka topic to write the data records to
+   * @param records            Data records to write to Kafka
+   * @param topologyTestDriver The {@link TopologyTestDriver} to send the data records to
+   * @param keySerializer      The {@link Serializer} corresponding to the key type
+   * @param valueSerializer    The {@link Serializer} corresponding to the value type
+   * @param <K>                Key type of the data records
+   * @param <V>                Value type of the data records
+   */
+  static <K, V> void produceKeyValuesSynchronously(final String topic,
+                                                   final List<KeyValue<K, V>> records,
                                                    final TopologyTestDriver topologyTestDriver,
                                                    final Serializer<K> keySerializer,
                                                    final Serializer<V> valueSerializer) {
-    for (final KeyValue<K, V> entity : entities) {
-      topologyTestDriver.pipeInput(new ConsumerRecord<>(
-        inputTopic,
+    for (final KeyValue<K, V> entity : records) {
+      final ConsumerRecord<byte[], byte[]> consumerRecord = new ConsumerRecord<>(
+        topic,
         0,
         0,
```

Comment:
Should we really set offset zero for all records? Maybe add a static Map from topic -> offset for tracking? (Might require a `reset()` method.)

Or let user pass in start offsets and increment?

Reply:
Not sure. I agree it creates some semantic ambiguity, although it apparently doesn't affect the logic under test. I guess Streams doesn't introspect on the offset, it just expects to get all the records in offset order. If we implement async processing, this may well change.

If it doesn't matter for this test, I think maybe we should just not do anything fancy. If it starts to matter at some point, I think we should revisit this util method then.

Reply:
Fair enough.